metaflow/plugins/argo/argo_workflows.py

```diff
@@ -574,6 +584,10 @@ def _compile_workflow_template(self):
             "metaflow/user": "argo-workflows",
             "metaflow/flow_name": self.flow.name,
         }
+
+        if self.parameters:
+            annotations.update({"metaflow/parameters": json.dumps(self.parameters)})
```

Comment:
Serialize the parameters dictionary into the `metaflow/parameters` annotation for consumers to deserialize.